# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/CanvasClient.cpp
# Commit: 6f19f43e0a0b
# Full Hash: 6f19f43e0a0bad72fa370191f4cc7f86f58566c2
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 09:34:54
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - NON_PREMULT TextureClient iff NON_PREMULT CanvasClient. r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78800
# ==============================================================================

diff -r 073302d26c5e -r 6f19f43e0a0b gfx/layers/client/CanvasClient.cpp
--- a/gfx/layers/client/CanvasClient.cpp	Thu Jun 11 06:37:38 2020 +0000
+++ b/gfx/layers/client/CanvasClient.cpp	Thu Jun 11 06:37:40 2020 +0000
@@ -31,6 +31,12 @@
 void CanvasClient::UseTexture(TextureClient* const aTexture) {
   MOZ_ASSERT(aTexture);
 
+  const auto isClientNonPremult =
+      bool(mTextureFlags & TextureFlags::NON_PREMULTIPLIED);
+  const auto isTextureNonPremult =
+      bool(aTexture->GetFlags() & TextureFlags::NON_PREMULTIPLIED);
+  MOZ_ALWAYS_TRUE(isTextureNonPremult == isClientNonPremult);
+
   bool changed = false;
 
   if (aTexture != mFrontBuffer) {
